Source/Projections.Store/Definition/OccurredFormat.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Dolittle.Runtime.Rudimentary;
+
+namespace Dolittle.Runtime.Projections.Store.Definition;
+
+/// <summary>
+/// Represents the format used to represent the date time mapping of when events occurred to projections.
+/// </summary>
+/// <param name="Value">The key selector expression as a string.</param>
+public record OccurredFormat(string Value) : ConceptAs<string>(Value)
+{
+    /// <summary>
+    /// Implicit operator from string.
+    /// </summary>
+    /// <param name="key">The occurred format.</param>
+    public static implicit operator OccurredFormat(string key) => new(key);
```

Comment:
In stead of the validator, we could also do it here in the concept. There could be an IsValid method that checks, that is used to check the registration arguments - and then in the constructor here we could throw if IsValid returns false. That way, it shouldn't be possible to sneakily create an invalid format anywhere.

Reply:
Problem with that is that it's "allowed" that it's empty or null when the KeySelectorType is not Occurred, see how the KeySelector instances are created with empty strings for the fields that are not used